analytics/views/stats.py

```diff
@@ -241,6 +241,7 @@ def get_chart_data(
     remote: bool = False,
     remote_realm_id: Optional[int] = None,
     server: Optional["RemoteZulipServer"] = None,
+    stream_id: Optional[int] = REQ(converter=to_non_negative_int, default=None),
```

Comment:
This logic as written did no access control check for the user having access to the stream, which would have been a security bug. I rewrote this part of the system to address that.